### PR TITLE
[Feature] Exercise dashboard button in exercise details pages

### PR DESCRIPTION
**Pull request by SimonEntholzer**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all newly inserted strings into English and German.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).

### Motivation and Context

In the exercise details pages I added a button to directly access the exercise-assessment dashboard.
Furthermore I cleaned up the button layout a bit. There is still some room for further improvements.

### Steps for Testing

1. Navigate to the course oveview -> exercises
2. Click on an exercise of every type (directly on the name), and check out the exercise detail page (Text, Modeling, Programming, File-Upload)
3. Test every button on those pages
4. Now, in an exam with a programming exercise, check the exercise details page too.

### Screenshots

File-Upload-exercise details page:
image
Programming exercise details page:
image
Modeling & text exercise details page: (they are identical)
image

### CODE REVIEW

**Review by willergeorg — APPROVED** [review]
Tested on TS1. Worked for me.

**Review by ga62suz — CHANGES_REQUESTED** [review]
Tested on TS2, with instructor account.
Everything works fine except the following:

Programming exercise:

Manage Hints doesn't seem to do anything

Newly created programming exercises doesn't have a dashboard button
image

Some old programming exercises also doesn't have a dashboard button, even though they have multiple graded submissions
image

Recreate Build Plans does not seem to trigger a rebuild of Template Result (reset the timer)
image

there is a programming exercise with no Update Structure Test Oracle button (KotlinExercise)
image

**Review by argertboja — COMMENTED** [review]
Tested on TS1, everything looks good!

**Review by canyelten — APPROVED** [review]
Tested on TS1, works as expected.

**Comment by SimonEntholzer** [comment]
> Tested on TS2, with artemis_test_user_11.
> Everything works fine except the following:
> 
> Programming exercise:
> 
> Manage Hints doesn't seem to do anything

Are you sure? For me it works. Which exercise did you use, specifically?

> Newly created programming exercises doesn't have a dashboard button
> 

This is intended, as the button is only shown when the manual assessment for the exercise is enabled. This is option comes available as soon as you set a due date.

> Some old programming exercises also doesn't have a dashboard button, even though they have multiple graded submissions
> 
Again, probably manual assessment is not enabled.

> Recreate Build Plans does not seem to trigger a rebuild of Template Result (reset the timer)
> 
I have to look into that, I'm not sure in this case tbh.

> there is a programming exercise with no Update Structure Test Oracle button (KotlinExercise)
> 
This is intended, the button is only shown for Java exercises.

**Comment by ga62suz** [comment]
> > Tested on TS2, with artemis_test_user_11.
> > Everything works fine except the following:
> > Programming exercise:
> > Manage Hints doesn't seem to do anything
> 
> Are you sure? For me it works. Which exercise did you use, specifically?
All of the exercises I tested was like that, but it could be because artemis_test_user_11 was not granted Instructor permission, which I did not realize until later (artemis_test_user_11 is noted as Instructor in the test credentials page)
 
> > Recreate Build Plans does not seem to trigger a rebuild of Template Result (reset the timer)
> > 
> > I have to look into that, I'm not sure in this case tbh.
Perhaps the problem lies with the triggering account's permission too.

**Review by NicolasUllmann — DISMISSED** [review]
Tested on TS2. Everything worked as expected.

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts [hunk 1]
```diff
@@ -484,6 +479,9 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
             },
         );
     }
+    onClick() {
+        console.log("asdfadsfdsf");
+    }
```
Comment (Olimbia23) [inline src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts:484]: Nice change but can we remove it hahah 😄

File: src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts [hunk 2]
```diff
@@ -92,6 +94,14 @@ export class ProgrammingExerciseDetailComponent implements OnInit, OnDestroy {
             this.statisticsService.getExerciseStatistics(programmingExercise.id).subscribe((statistics: ExerciseManagementStatisticsDto) => {
                 this.doughnutStats = statistics;
             });
+            if(!this.isExamExercise) {
+                this.baseResource =  `/course-management/${this.courseId}/programming-exercises/${programmingExercise.id}/`;
```
Comment (Olimbia23) [inline src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts:98]: Please don't forget to address the `client-style` issues.

**Review by Olimbia23 — COMMENTED** [review]
Solid work @SimonEntholzer. Left some minor comments.

**Review by katjanakosic — DISMISSED** [review]
Tested on TS1, worked fine 👍

**Review by cremertim — DISMISSED** [review]
tested on ts1.. works as expected

**Review by sinangunduz — DISMISSED** [review]
tested on ts1, works as expected

**Comment by imstreetspirit** [comment]
tested on ts2 works as expected. code seem good in general, there are only minor client-style issues need to be fixed.... 👍

**Review by ivohashamov — DISMISSED** [review]
Code is looking good 👍

**Review by ArtemTsurikov — APPROVED** [review]
Tested on ts1, worked fine.

**Pull request merged**